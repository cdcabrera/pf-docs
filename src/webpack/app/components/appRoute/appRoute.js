import React from 'react';
import { useLocation } from '@reach/router';
import { Footer } from '../footer/footer';

export const AppRoute = ({ child, katacodaLayout, title, hasFooter = process.env.hasFooter }) => {
  const location = useLocation();

  if (window?.gtag) {
    window.gtag('config', 'UA-47523816-6', {
      page_path: location.pathname,
      page_title: title || location.pathname
    });
  }

  return (
    <React.Fragment>
      {child}
      {!katacodaLayout && hasFooter && <Footer />}
    </React.Fragment>
  );
};
