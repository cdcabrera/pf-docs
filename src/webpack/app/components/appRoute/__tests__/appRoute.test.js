import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { AppRoute } from '../appRoute';

describe('AppRoute', () => {
  it('should render a basic component', () => {
    const { container } = render(<AppRoute child="Lorem ipsum" />);
    expect(container).toMatchSnapshot('basic');
  });
});
