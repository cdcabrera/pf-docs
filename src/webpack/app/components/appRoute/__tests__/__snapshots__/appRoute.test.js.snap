// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoute should render a basic component: basic 1`] = `
<div>
  Lorem ipsum
</div>
`;
